src/core/Xtarget.ml

```diff
@@ -8,9 +8,36 @@
  *)
 
 type t = {
-  file : Fpath.t;
+  source : Source.t;
+  internal_path_to_content : Fpath.t;
```

Comment:
again ... an internal_path_to_content and a source. Bad smell.

Reply:
Reorg in fc35371b4 addresses this somewhat.